nowplaying/cli.py

```diff
@@ -89,16 +89,19 @@ def signal_handler(self, signum, frame):
             sys.exit(os.EX_OK)
 
     def setup_logging(self):
-        logger.setLevel(logging.DEBUG)
+        root = logging.getLogger()
```

Comment:
* Isn't the logger object already available?
* Please use a more descriptive name than `root`.

Reply:
The logger never gets properly introduced in this scope here. The only place where the root logger was accessed to make it available in this scope was way down below in the `__main__` block where it basically introduced the logger to the NowPlaying class as a side effect from the outer scope.

https://github.com/radiorabe/nowplaying/blob/61ebd9db974da4ec2b5b1cb8658922f2994ed781/nowplaying/cli.py#L260

The [python 3 logging cookbook](https://docs.python.org/3/howto/logging-cookbook.html) has heaps of examples where `root` is used as the variable name for the root logger. I consider the variable name rather pythonic.

Reply:
> The logger never gets properly introduced in this scope here. The only place where the root logger was accessed to make it available in this scope was way down below in the `__main__` block where it basically introduced the logger to the NowPlaying class as a side effect from the outer scope.
> 
> https://github.com/radiorabe/nowplaying/blob/61ebd9db974da4ec2b5b1cb8658922f2994ed781/nowplaying/cli.py#L260
> 
> The [python 3 logging cookbook](https://docs.python.org/3/howto/logging-cookbook.html) has heaps of examples where `root` is used as the variable name for the root logger. I consider the variable name rather pythonic.

There are also many examples which use `rootLogger` within your referenced cookbook. Which is IMHO much easier to understand while reading the code .

Reply:
> Exactly, so it's already available then. If you think it's more appropriate to initialize it within this method directly (I'm not sure about scoping)...

Using a module scoped logger is fine if it's used in its core logger way (ie. to call log methods) as that pattern is widely recognized. When we want to work on the logging object composition (ie. by using `addHandler`) I feel it is much more sane to work on a local copy of the root logger singleton.

> There are also many examples which use rootLogger within your referenced cookbook. 

I count 6 instances of `root = logging.getLogger ` and just one `rootLogger = logging.getLogger`.

Reply:
A further distinction to note is also `logging.getLogger()` during setup vs. the copy from `logging.getLogger("now-playing")` we use in the module's scope.

Reply:
So, we won't use the same logger instance and configuration then?

Reply:
This configures the root logger, all other loggers always take the root logger configuration into account as they use the root logger under the hood. This really is how most of the examples in the [cookbook](https://docs.python.org/3/howto/logging-cookbook.html) work.